seatunnel-connectors-v2/connector-http/connector-http-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/http/source/HttpSourceFactory.java

```diff
@@ -49,7 +61,6 @@ public OptionRule.Builder getHttpBuilder() {
                 .optional(HttpConfig.FORMAT)
                 .optional(HttpConfig.JSON_FIELD)
                 .optional(HttpConfig.CONTENT_FIELD)
-                .conditional(HttpConfig.METHOD, HttpRequestMethod.POST, HttpConfig.BODY)
```

Comment:
Why remove this?

Reply:
Because when using the param parameter, the post method is also used without a body

Reply:
Yes, And you need add `HttpConfig.BODY` use `.optional(HttpConfig.BODY)`. Otherwise, the body parameter will be lost.

Reply:
done PTAL @EricJoy2048